packages/slate/src/transforms/node.ts

```diff
@@ -940,8 +940,14 @@ export const NodeTransforms: NodeTransforms = {
         if (matches.length > 0) {
           const [first] = matches
           const last = matches[matches.length - 1]
-          const [, firstPath] = first
-          const [, lastPath] = last
+          const [firstNode, firstPath] = first
+          const [lastNode, lastPath] = last
+
+          if (firstNode === lastNode && Editor.isEditor(firstNode)) {
```

Comment:
Wouldn't this just be `if (firstPath.length === 0 && lastPath.length === 0)`? Since the only way you would ever get editor in a nodePath pair is if the path is an empty path. Would give a slight performance boost since isEditor checks nodes etc. Would be more hard to read though so maybe a comment would make it clearer.